Projects/Server/GameCounter.h

```diff
@@ -0,0 +1,43 @@
+#ifndef GAMECOUNTER_H
+#define GAMECOUNTER_H
+
+#include "State.h"
+#include "GameHandler.h"
+
+#include <map>
+
+
+
+class GameCounter {
+public:
+	GameCounter(const AbstractPlayer::Ptr& player0, const AbstractPlayer::Ptr& player1, const size_t& gameSample);
+	
+	friend std::ostream& operator<<(std::ostream& os, const GameCounter& gameCounter);
+
+	void play();
+
+	size_t getP0Win() const;
+	size_t getP1Win() const;
+	size_t getDraw() const;
+	size_t getGamesPlayed() const;
+	size_t getGameSample() const;
+	AbstractPlayer::Ptr getPlayer0() const;
+	AbstractPlayer::Ptr getPlayer1() const;
+	std::string getPlayer0Name() const;
+	std::string getPlayer1Name() const;
+	
+private:
+	AbstractPlayer::Ptr _player0;
+	AbstractPlayer::Ptr _player1;
+	std::map<std::string, size_t> gameStats;
```

Comment:
Avoid [stringly-typed](https://wiki.c2.com/?StringlyTyped) parameters.
You can use `State::Outcome` for the key type here.

Reply:
You can even separate the Outcome-related part into its own map, and keep `_gameSample` and `gamesPlayed` as separate member fields.

Reply:
`std::string` is replaced by `State::Outcome` in the map `gameStats`. `_gameSample` and `gamesPlayed` have been taken out of the map to add as new data members.